rust/src/lib.rs

```diff
@@ -156,24 +136,31 @@ impl Anime {
                         } else {
                             0.0
                         };
-                        // add 1 for R indexing
                         // ensures that no duplicates are inserted. Creates a new empty vector is needed
-                        let entry = matches.entry(j as i32).or_default();
+                        let entry = matches.entry(j).or_default();
 
-                        if let Some(tuple) = entry.iter_mut().find(|x| x.index == i as i32) {
+                        if let Some(tuple) = entry.iter_mut().find(|x| x.source_index == i) {
                             tuple.shared_len += shared_len;
                         } else {
-                            entry.extend(std::iter::once(MatchCandidate {
-                                index: i as i32,
+                            entry.push(MatchCandidate {
+                                source_index: i,
                                 shared_len,
-                            }));
+                            });
                         }
                     }
                 }
             }
         });
-        self.matches.set(matches)?;
-        Ok(self)
+
+        Self {
+            distance_tolerance,
```

Comment:
If some of these fields aren't used for interpolation, arguably they could be removed from the struct, unless it's potentially helpful

Reply:
they are.